snapshot/git/gitdb/backends.go

```diff
@@ -0,0 +1,65 @@
+package gitdb
+
+import (
+	"fmt"
+	"strings"
+
+	snap "github.com/scootdev/scoot/snapshot"
+)
+
+// backend allows getting a snapshot for an ID, which can then be used to download the ID
+type backend interface {
+	parseID(id string, kind string, parts []string) (snapshot, error)
```

Comment:
should kind be constrained to a 'backendType' struct or interface? (be more explicit about what is allowed)